src/rimeengine.cpp

```diff
@@ -316,7 +316,14 @@ void RimeEngine::rimeStart(bool fullcheck) {
             RIME_ERROR() << "Failed to create user directory: " << userDir;
         }
     }
+#ifdef __ANDROID__
```

Comment:
initialize sharedDataDir only once in constructor.

Also, if it's not found, throw an exception in constructor.